js/jquery.mapael.js

```diff
@@ -630,6 +635,164 @@
+         *  opt.afterShowRange callback
+         *  opt.ranges the range to show:
+         *  Example:
+         *  opt.ranges = {
+         *      'plot' : {
+         *          0 : {                        // valueIndex
+         *              'min': 1000,
+         *              'max': 1200
+         *          },
+         *          1 : {                        // valueIndex
+         *              'min': 10,
+         *              'max': 12
+         *          }
+         *      },
+         *      'area' : {
+         *          {'min': 10, 'max': 20}    // No valueIndex, only an object, use 0 as valueIndex (easy case)
+         *      }
+         *  }
+         */
+        onShowElementsInRange: function(e, opt) {
+            var self = this;
+
+            // set animDuration to default if not defined
+            if (opt.animDuration === undefined) {
+                opt.animDuration = 0;
+            }
+
+            // set hiddenOpacity to default if not defined
+            if (opt.hiddenOpacity === undefined) {
+                opt.hiddenOpacity = 0.3;
+            }
+
+            // handle area
+            if (opt.ranges && opt.ranges.area) {
+                self.showElemByRange(opt.ranges.area, self.areas, opt.hiddenOpacity, opt.animDuration);
+            }
+
+            // handle plot
+            if (opt.ranges && opt.ranges.plot) {
+                self.showElemByRange(opt.ranges.plot, self.plots, opt.hiddenOpacity, opt.animDuration);
+            }
+
+            // handle link
+            if (opt.ranges && opt.ranges.link) {
+                self.showElemByRange(opt.ranges.link, self.links, opt.hiddenOpacity, opt.animDuration);
+            }
+
+            // Call user callback
+            if (opt.afterShowRange) opt.afterShowRange();
+        },
+
+        /*
+         * Show some element in range
+         * @param ranges: the ranges
+         * @param elems: list of element on which to check against previous range
+         * @hiddenOpacity: the opacity when hidden
+         * @animDuration: the animation duration
+         */
+        showElemByRange: function(ranges, elems, hiddenOpacity, animDuration) {
+            var self = this;
+            // Hold the final opacity value for all elements consolidated after applying each ranges
+            // This allow to set the opacity only once for each elements
+            var elemsFinalOpacity = {};
+
+            // set object with one valueIndex to 0 if we have directly the min/max
+            if (ranges.min !== undefined || ranges.max !== undefined) {
+                ranges = {0: ranges};
+            }
+
+            // Loop through each valueIndex
+            $.each(ranges, function (valueIndex) {
+                var range = ranges[valueIndex];
+                // Check if user defined at least a min or max value
+                if (range.min === undefined && range.max === undefined) {
+                    return true; // next range
```

Comment:
Actually, no. This `return true;` is related to the `$.each()` function callback.
Returning true will go to the next item (like a `continue` statement). Whereas false (or undefined) will end the loop (much like a `break` statement). See jQuery doc: http://api.jquery.com/jQuery.each/